src/hotspot/share/jeandle/templatemodule/jeandleRuntimeDefinedJavaOps.cpp

```diff
@@ -103,6 +103,34 @@ DEF_JAVA_OP(safepoint_poll, 1, llvm::Type::getVoidTy(context))
   ir_builder.CreateRetVoid();
 JAVA_OP_END
 
+DEF_JAVA_OP(newarray, 0, llvm::PointerType::get(context, llvm::jeandle::AddrSpace::JavaHeapAddrSpace),
+            llvm::Type::getInt32Ty(context),  // length
+            llvm::Type::getInt32Ty(context))  // type
+
+    llvm::Value* length = func->getArg(0);
+    llvm::Value* type = func->getArg(1);
+
+    // Get current thread pointer using jeandle.current_thread JavaOp
+    llvm::Function* current_thread_func = template_module.getFunction("jeandle.current_thread");
+    if (!current_thread_func) {
+        RuntimeDefinedJavaOps::set_failed("jeandle.current_thread is not found in template module");
+        return;
+    }
+    llvm::CallInst* current_thread = ir_builder.CreateCall(current_thread_func);
+    current_thread->setCallingConv(llvm::CallingConv::Hotspot_JIT);
+
+    llvm::CallInst* call_inst = ir_builder.CreateCall(JeandleRuntimeRoutine::new_typeArray_callee(template_module), {type, length, current_thread});
+    call_inst->setCallingConv(llvm::CallingConv::Hotspot_JIT);
+
+    // Load result from vm_result
+    llvm::Type* oop_type = llvm::PointerType::get(context, llvm::jeandle::AddrSpace::JavaHeapAddrSpace);
+    llvm::Value* vm_result_offset = ir_builder.getInt64(static_cast<uint64_t>(JavaThread::vm_result_offset()));
+    llvm::Value* vm_result_addr = ir_builder.CreateGEP(ir_builder.getInt8Ty(), current_thread, vm_result_offset);
+    llvm::Value* vm_result_ptr = ir_builder.CreateBitCast(vm_result_addr, llvm::PointerType::get(oop_type, 0));
+    llvm::Value* result = ir_builder.CreateLoad(oop_type, vm_result_ptr);
+    ir_builder.CreateRet(result);
```

Comment:
Loading the result oop can be implemented in `JeandleCallVM`, thus all other runtime routines returning oop values can reuse this code.
Refer to `opto/generateOptoStub.cpp` for more details.